### DISCUSSION TITLE
how to check if the user's access_token is not expired with google auth

### DISCUSSION THREAD

**Discussion by leowenlu** [fastapi-users/fastapi-users#1447] [Q&A]
my setup:
1. using `Bearer`
2. have added/included google_oauth_client
```
    app.include_router(
        fastapi_users.get_oauth_router(
            google_oauth_client, 
            auth_backend, 
            SECRET,
            associate_by_email=True,
            redirect_url=redirect_url
        ),
        prefix=f"{CONFIG.API_PREFIX}/auth/google",
        tags=["auth"],
    )
```
3. frontend is using react.

what I want to achieve or know:

1. the access_token validate time range,
2. how to refresh token
so I can redirect the user to login page if needed.

**Comment by hgalytoby** [comment]
> my setup:
> 
> 1. using `Bearer`
> 2. have added/included google_oauth_client
> 
> ```
>     app.include_router(
>         fastapi_users.get_oauth_router(
>             google_oauth_client, 
>             auth_backend, 
>             SECRET,
>             associate_by_email=True,
>             redirect_url=redirect_url
>         ),
>         prefix=f"{CONFIG.API_PREFIX}/auth/google",
>         tags=["auth"],
>     )
> ```
> 
> 3. frontend is using react.
> 
> what I want to achieve or know:
> 
> 1. the access_token validate time range,
> 2. how to refresh token
>    so I can redirect the user to login page if needed.


This example sets the access_token time.
https://fastapi-users.github.io/fastapi-users/latest/configuration/authentication/backend/#create-a-backend

I'm not sure if you're trying to update a third-party token or a fastapi-users token, but to refresh the token you need to write a new method to implement the

**Comment by leowenlu** [comment]
@hgalytoby 

Thanks for the link.
I understand that `JWTStrategy(secret=SECRET, lifetime_seconds=3600)` will allow me to set time range.
For refresh the token, I found this is quite helpful, and looking into it now.
https://github.com/fastapi-users/fastapi-users/pull/1367#issuecomment-2431854784
